vm/loader.go

```diff
@@ -64,7 +64,10 @@ func (l *Loader) LoadProgs(programPath string) error {
 			if fi.IsDir() {
 				continue
 			}
-			l.LoadProg(path.Join(programPath, fi.Name()))
+			err = l.LoadProg(path.Join(programPath, fi.Name()))
+			if err != nil {
+				return err
```

Comment:
this should capture the error in a slice of errors, and return them all after the loop is complete.